src/main/java/pl/betoncraft/betonquest/compatibility/brewery/GiveBrewEvent.java

```diff
@@ -0,0 +1,68 @@
+package pl.betoncraft.betonquest.compatibility.brewery;
+
+import com.dre.brewery.BIngredients;
+import com.dre.brewery.BRecipe;
+import org.bukkit.entity.Player;
+import pl.betoncraft.betonquest.Instruction;
+import pl.betoncraft.betonquest.InstructionParseException;
+import pl.betoncraft.betonquest.QuestRuntimeException;
+import pl.betoncraft.betonquest.api.QuestEvent;
+import pl.betoncraft.betonquest.utils.PlayerConverter;
+
+public class GiveBrewEvent extends QuestEvent{
+
+    private Integer amount;
+    private Integer quality;
+    private BRecipe recipe;
+
+    public GiveBrewEvent(Instruction instruction) throws InstructionParseException {
+        super(instruction);
+
+        amount = instruction.getInt();
+
+        quality = instruction.getInt();
+
+        if(quality < 0 || quality > 10){
+            throw new InstructionParseException("Brew quality must be between 0 and 10!");
+        }
+
+        String name = instruction.next().replace("_", " ");
+
+        BRecipe recipe = null;
+        for(BRecipe r : BIngredients.recipes){
+            if(r.hasName(name)){
+                recipe = r;
+                break;
+            }
+        }
+
+        if(recipe == null){
+            throw new InstructionParseException("There is no brewing recipe with the name " + name + "!");
+        }else{
+            this.recipe = recipe;
+        }
+    }
+
+    @Override
+    public void run(String playerID) throws QuestRuntimeException {
+        Player p = PlayerConverter.getPlayer(playerID);
+
+        int remaining = amount;
+
+        for(int i = 0; i < p.getInventory().getSize(); i++){
+            if(p.getInventory().getItem(i) == null){
+                p.getInventory().setItem(i, recipe.create(quality));
```

Comment:
![2017-11-01_13 00 18](https://user-images.githubusercontent.com/6470178/32273980-116c413e-bf05-11e7-8d5f-5f1cd3c225f4.png)
There's a nice way to do this using `player.getInventory().addItem(ItemStack... items)`. You could create an array of brews (since we can't add a single stack with size greater than 1) and use that method. It will return a map of ItemStacks that didn't fit. You can iterate over that map to drop excess brews on the ground.

This method won't put a bottle on player's head :laughing:

Reply:
Hey, but that's a great stunt at parties :D
Yes, I'll fix it. Turns out writing a quick solution and then having to revise it as problems occour takes longer than thinking about it a little more before writing any code...

Reply:
It's not aa bug it's a feature 😂

Reply:
Think twice, code once 😄 But I had a good laugh when I saw that.